moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -111,6 +166,79 @@ export function useMapDrawTools() {
     []
   );
 
+  /**
+   * Add existing drawn points in the project extent feature collection to the draw UI so they are editable
+   */
+  const initializeExistingDrawFeatures = useCallback(
+    ref => {
+      if (ref) {
+        // Only add features that are not already present in the draw UI to avoid duplicates
+        const drawnFeatures = getDrawnFeaturesFromFeatureCollection(
+          featureCollection
+        );
+        const featuresAlreadyInDrawMap = ref.getFeatures();
+
+        const featuresToAdd = findDifferenceByFeatureProperty(
+          "PROJECT_EXTENT_ID",
+          drawnFeatures,
+          featuresAlreadyInDrawMap
+        );
+
+        ref.addFeatures(featuresToAdd);
+      }
+    },
+    [featureCollection]
+  );
```

Comment:
This is pretty intense code, the useCallback returns a function that adds only new features to a component?